app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/metrics_capture.rb

```diff
@@ -0,0 +1,151 @@
+      :rollup                => "average",
+      :unit_key              => "kilobytespersecond",
+      :capture_interval_name => "realtime",
+    }.freeze,
+    "net_usage_rate_average"     => {
+      :counter_key           => "net_usage_rate_average",
+      :instance              => "",
+      :capture_interval      => "20",
+      :precision             => 2,
+      :rollup                => "average",
+      :unit_key              => "kilobytespersecond",
+      :capture_interval_name => "realtime",
+    }.freeze,
+    "mem_usage_absolute_average" => {
+      :counter_key           => "mem_usage_absolute_average",
+      :instance              => "",
+      :capture_interval      => "20",
+      :precision             => 1,
+      :rollup                => "average",
+      :unit_key              => "percent",
+      :capture_interval_name => "realtime",
+    }.freeze,
+  }.freeze
+
+  def perf_collect_metrics(interval_name, _start_time = nil, _end_time = nil)
+    require 'rest-client'
+
+    raise _("No EMS defined") if ext_management_system.nil?
+
+    counters_by_mor = {target.ems_ref => VIM_STYLE_COUNTERS}
+    counter_values_by_mor = {target.ems_ref => {}}
+
+    metrics_endpoint = ext_management_system.endpoints.find_by(:role => "metrics")
+    raise _("Missing monitoring instance id") if metrics_endpoint.nil?
+
+    instance_id = metrics_endpoint.options["monitoring_instance_id"]
+
+    response = RestClient::Request.execute(
+      :method  => :post,
+      :url     => "https://#{ext_management_system.provider_region}.monitoring.cloud.ibm.com/api/data",
+      :headers => {
+        'Content-Type'  => 'application/json',
+        'Authorization' => "Bearer #{iam_access_token}",
+        'IBMInstanceID' => instance_id
+      },
+      :payload => JSON.generate(get_metrics_query(target.name))
+    )
+    data = JSON.parse(response.body)
+    dataset = consolidate_data(data["data"])
+
+    store_datapoints!(dataset[:timestamps], dataset[:cpu_usage_rate_average], "cpu_usage_rate_average", counter_values_by_mor[target.ems_ref])
+    store_datapoints!(dataset[:timestamps], dataset[:mem_usage_absolute_average], "mem_usage_absolute_average", counter_values_by_mor[target.ems_ref])
+    store_datapoints!(dataset[:timestamps], dataset[:net_usage_rate_average], "net_usage_rate_average", counter_values_by_mor[target.ems_ref])
+    store_datapoints!(dataset[:timestamps], dataset[:disk_usage_rate_average], "disk_usage_rate_average", counter_values_by_mor[target.ems_ref])
+
+    return counters_by_mor, counter_values_by_mor
+  rescue RestClient::ExceptionWithResponse => err
+    log_header = "[#{interval_name}] for: [#{target.class.name}], [#{target.id}], [#{target.name}]"
+    _log.error("#{log_header} Unhandled exception during perf data collection: [#{err}], class: [#{err.class}]")
+    _log.log_backtrace(err)
+    raise
+  end
+
+  def get_metrics_query(instance_name)
+    {
+      :last           => 144_00,
+      :sampling       => 600,
```

Comment:
@nasark can you add a comment describing what `:last` and `:sampling` here are?  I assume `:sampling => 600` means last ten minutes but then what is `:last` ?  Also think you want to break up `14,000` as `14_000` not `140_00`

Reply:
Sure, I'll add a comment. `last` is referring to retrieving data from the last 4 hours (14400 seconds). `sampling` specifies the data resolution and so a single data point (aggregated value) every 10 minutes. There's no particular reason why I chose these parameters, open to suggestions

Reply:
Ah okay, so we have `start_time` param which will give you the "last" effectively.  These are optional but other providers like amazon do the following:

```
def perf_collect_metrics(interval_name, start_time = nil, end_time = nil)
  end_time ||= Time.now
  end_time   = end_time.utc
  start_time ||= end_time - 4.hours
  start_time   = start_time.utc
```

You can then use this to figure out the `:last` seconds

As for sampling I think one-per-minute would be preferable to one-per-ten-minutes